### ISSUE TITLE
operator crashes with: "assignment to entry in nil map"

### ISSUE THREAD

**Issue by marsom** [blackducksoftware/synopsys-operator#1037]
**What happened**: createing a blackduck instance causees the synopis operator to crash

**How to reproduce it (as minimally and precisely as possible)**:
- install 2019.8.4
- deploy a blackduck instance

**Screenshots/additional context**:

```
time="2019-10-28T06:59:41Z" level=debug msg="patch namespace for synopsys label in namespace 'appl-blackduck-e1'"
E1028 06:59:41.847036       1 runtime.go:69] Observed a panic: "assignment to entry in nil map" (assignment to entry in nil map)
/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190313205120-d7deff9243b1/pkg/util/runtime/runtime.go:76
/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190313205120-d7deff9243b1/pkg/util/runtime/runtime.go:65
/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190313205120-d7deff9243b1/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/panic.go:522
/usr/local/go/src/runtime/map_faststr.go:204
/go/src/github.com/blackducksoftware/synopsys-operator/pkg/crdupdater/namespace.go:132
/go/src/github.com/blackducksoftware/synopsys-operator/pkg/crdupdater/namespace.go:89
/go/src/github.com/blackducksoftware/synopsys-operator/pkg/crdupdater/updater.go:73
/go/src/github.com/blackducksoftware/synopsys-operator/pkg/crdupdater/crudcomponents.go:188
/go/src/github.com/blackducksoftware/synopsys-operator/pkg/apps/blackduck/latest/creater.go:105
/go/src/github.com/blackducksoftware/synopsys-operator/pkg/apps/blackduck/blackduck.go:193
/go/src/github.com/blackducksoftware/synopsys-operator/pkg/blackduck/crdhandler.go:187
/go/src/github.com/blackducksoftware/synopsys-operator/pkg/blackduck/crdhandler.go:107
/go/src/github.com/blackducksoftware/synopsys-operator/pkg/blackduck/crdcontroller.go:153
/go/src/github.com/blackducksoftware/synopsys-operator/pkg/blackduck/crdcontroller.go:89
/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190313205120-d7deff9243b1/pkg/util/wait/wait.go:152
/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190313205120-d7deff9243b1/pkg/util/wait/wait.go:153
/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190313205120-d7deff9243b1/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:1337
panic: assignment to entry in nil map [recovered]
        panic: assignment to entry in nil map

goroutine 92 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190313205120-d7deff9243b1/pkg/util/runtime/runtime.go:58 +0x105
panic(0x1692180, 0x1ab8fe0)
        /usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/blackducksoftware/synopsys-operator/pkg/crdupdater.(*Namespace).patch(0xc0003e2770, 0x183c0e0, 0xc00017ac60, 0xc00017ac00, 0x0, 0x0, 0xc000616c40)
        /go/src/github.com/blackducksoftware/synopsys-operator/pkg/crdupdater/namespace.go:132 +0x6f0
github.com/blackducksoftware/synopsys-operator/pkg/crdupdater.(*Namespace).add(0xc0003e2770, 0x0, 0x0, 0x4423a2, 0xc00035c100)
        /go/src/github.com/blackducksoftware/synopsys-operator/pkg/crdupdater/namespace.go:89 +0xb5d
github.com/blackducksoftware/synopsys-operator/pkg/crdupdater.(*Updater).Update(0xc000556fc8, 0x0, 0x0, 0x0)
        /go/src/github.com/blackducksoftware/synopsys-operator/pkg/crdupdater/updater.go:73 +0x96
github.com/blackducksoftware/synopsys-operator/pkg/crdupdater.(*CommonConfig).CRUDComponents(0xc000114620, 0xc0003a0500, 0xc0002a0000, 0xc0002f2ea0, 0x11)
        /go/src/github.com/blackducksoftware/synopsys-operator/pkg/crdupdater/crudcomponents.go:188 +0x12af
github.com/blackducksoftware/synopsys-operator/pkg/apps/blackduck/latest.(*Creater).Ensure(0xc0005eb830, 0xc00016a800, 0x8, 0x1b16000)
        /go/src/github.com/blackducksoftware/synopsys-operator/pkg/apps/blackduck/latest/creater.go:105 +0xe2b
github.com/blackducksoftware/synopsys-operator/pkg/apps/blackduck.(*Blackduck).Ensure(0xc000640940, 0xc00016a800, 0xc000640940, 0x1887d31)
        /go/src/github.com/blackducksoftware/synopsys-operator/pkg/apps/blackduck/blackduck.go:193 +0xb8
github.com/blackducksoftware/synopsys-operator/pkg/blackduck.(*Handler).ObjectUpdated(0xc0000a37c0, 0x0, 0x0, 0x1828e20, 0xc00016a800)
        /go/src/github.com/blackducksoftware/synopsys-operator/pkg/blackduck/crdhandler.go:187 +0x37e
github.com/blackducksoftware/synopsys-operator/pkg/blackduck.(*Handler).ObjectCreated(0xc0000a37c0, 0x1828e20, 0xc00016a800)
        /go/src/github.com/blackducksoftware/synopsys-operator/pkg/blackduck/crdhandler.go:107 +0x47
github.com/blackducksoftware/synopsys-operator/pkg/blackduck.(*Controller).processNextItem(0xc0000a3840, 0x0)
        /go/src/github.com/blackducksoftware/synopsys-operator/pkg/blackduck/crdcontroller.go:153 +0x2b3
github.com/blackducksoftware/synopsys-operator/pkg/blackduck.(*Controller).runWorker(0xc0000a3840)
        /go/src/github.com/blackducksoftware/synopsys-operator/pkg/blackduck/crdcontroller.go:89 +0xdd
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0003b0dc0)
        /go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190313205120-d7deff9243b1/pkg/util/wait/wait.go:152 +0x54
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0003b0dc0, 0x3b9aca00, 0x0, 0x1, 0xc0002de0c0)
        /go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190313205120-d7deff9243b1/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc0003b0dc0, 0x3b9aca00, 0xc0002de0c0)
        /go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190313205120-d7deff9243b1/pkg/util/wait/wait.go:88 +0x4d
created by github.com/blackducksoftware/synopsys-operator/pkg/blackduck.(*Controller).Run
        /go/src/github.com/blackducksoftware/synopsys-operator/pkg/blackduck/crdcontroller.go:71 +0x225
```

**Environment (please complete the following information):**
- Kubernetes version (use `kubectl version`): Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T19:09:08Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
- Cloud provider or hardware configuration: on premise
- OS (e.g: `cat /etc/os-release`): Red Hat Enterprise Linux Server release 7.6 (Maipo
- Kernel (e.g. `uname -a`): Linux e1-k8s-mlsu108 3.10.0-957.10.1.el7.x86_64 #1 SMP Thu Feb 7 07:12:53 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
- Others:

**Comment by marsom** [comment]
does not happen with 2019.8.3

**Comment by yashbhutwala** [comment]
@marsom thanks for reporting this.  Can you please specify the exact commands you used in order to reproduce your issue?

**Comment by marsom** [comment]
ok, i think in know the problem, it that the operator wants to update the namespace, however it its not allowd to do so.

**Comment by marsom** [comment]
`./synopsysctl create blackduck default \
        --namespace mynamsapce \
        --license-key x \
        --expose-ui NONE \
        --enable-binary-analysis \
        --enable-source-code-upload \
        --external-postgres-host x \
        --external-postgres-port 5434 \
        --external-postgres-ssl false \
        --external-postgres-admin blackduck \
        --external-postgres-admin-password x \
        --external-postgres-user blackduck \
        --external-postgres-user-password x \
        --proxy-certificate-file-path /path/to/pem \
        --environs IPV4_ONLY:1,BLACKDUCK_HUB_SERVER_ADDRESS:0.0.0.0,HUB_PROXY_SCHEME:https,HUB_PROXY_HOST:proxy.example.org,HUB_PROXY_PORT:3128 \
        --size small \
        --persistent-storage false`

**Comment by marsom** [comment]
it doesn't seam to be an rbac issue, the  role synopsys-operator-admin and the RoleBinding synopsys-operator-admin seams to be correct.

**Comment by yashbhutwala** [comment]
@marsom I think there is two things going on here: 1) the k8s version you're running on is `1.16` 2) the difference in our dependency between `2019.8.4` and `2019.8.3`: https://github.com/blackducksoftware/synopsys-operator/blob/2019.8.4/go.mod#L30

`2019.8.4` is compatible with `k8s<=v1.15`

**Comment by yashbhutwala** [comment]
@marsom actually never mind, the fix is coming in `2019.10.0` with PR #1028 

I didn't realize this is actually a duplicate of #1031
